numba/tests/test_numbers.py

```diff
@@ -0,0 +1,96 @@
+# (Some) Tests for numbers.py
+from __future__ import print_function, absolute_import, division
+
+import sys
+
+import numpy as np
+
+from numba import njit
+from numba.errors import TypingError
+from .support import TestCase, unittest
+
+_32bit_only = not (sys.maxsize > 2 ** 32)
+
+
+def gen_view(a,b):
+    def impl(x):
+        return a(x).view(b)
+    return impl
+
+
+class TestViewIntFloat(TestCase):
+
+    def do_testing(self, inputs, dtypes):
+        for value, initial_type, expected in inputs:
+            for target_type, result in zip(dtypes, expected):
+                view = njit(gen_view(initial_type, target_type))
+                if not np.isnan(result):
+                    self.assertEqual(view(value), target_type(result))
+                else:
+                    self.assertTrue(np.isnan(view(value)))
+
+    def test_8_bits(self):
+        dtypes = (np.uint8, np.int8)
+        #        Value  Initial Type   Expected answers using dtypes
+        inputs = ((1,   np.uint8,    (1, 1)),
+                  (-1,  np.int8,     (255, -1)))
+        self.do_testing(inputs, dtypes)
+
+    def test_32_bits(self):
+        dtypes = (np.uint32, np.int32, np.float32)
+        #        Value  Initial Type   Expected answers using dtypes
+        inputs = ((1,   np.uint32,    (1, 1, 1.401298464324817e-45)),
+                  (-1,  np.int32,     (4294967295, -1, np.nan)),
+                  (1.0, np.float32,   (1065353216, 1065353216, 1.0)))
+        self.do_testing(inputs, dtypes)
+
+    @unittest.skipIf(_32bit_only, "64 bit test only")
```

Comment:
Is this necessary? What's preventing this from working on 32bit?

Reply:
No, it's the other way around, it doesn't work on 32 bit systems because of the int64 types.

Reply:
I have a 32bit and a 64bit system locally, these tests pass on both?!

Reply:
Fixed with latest push.